src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -165,5 +172,585 @@ ConvSolution InitAnyInvokerFactory(const ProblemDescriptionType& problem,
+            {
+
+                return SolverType{ctx,
+                                  problem.GetInDataType(),
+                                  static_cast<uint32_t>(ck_args.N),
+                                  static_cast<uint32_t>(ck_args.C1),
+                                  static_cast<uint32_t>(ck_args.Di),
+                                  static_cast<uint32_t>(ck_args.Hi),
+                                  static_cast<uint32_t>(ck_args.Wi)};
+            }
+            else
+            {
+                return SolverType{ctx,
+                                  problem.GetInDataType(),
+                                  static_cast<uint32_t>(ck_args.N),
+                                  static_cast<uint32_t>(ck_args.C1),
+                                  static_cast<uint32_t>(ck_args.Hi),
+                                  static_cast<uint32_t>(ck_args.Wi)};
+            }
+        }
+        else if constexpr(CONV_OP_TAG == ConvOperandTag::Weights)
+        {
+            if constexpr(ND == 3)
+            {
+                return SolverType{ctx,
+                                  problem.GetWeightsDataType(),
+                                  static_cast<uint32_t>(ck_args.K1),
+                                  static_cast<uint32_t>(ck_args.C),
+                                  static_cast<uint32_t>(ck_args.Z),
+                                  static_cast<uint32_t>(ck_args.Y),
+                                  static_cast<uint32_t>(ck_args.X)};
+            }
+            else
+            {
+                return SolverType{ctx,
+                                  problem.GetWeightsDataType(),
+                                  static_cast<uint32_t>(ck_args.K1),
+                                  static_cast<uint32_t>(ck_args.C),
+                                  static_cast<uint32_t>(ck_args.Y),
+                                  static_cast<uint32_t>(ck_args.X)};
+            }
+        }
+        else
+        {
+            static_assert(CONV_OP_TAG == ConvOperandTag::Output);
+            if constexpr(ND == 3)
+            {
+                return SolverType{ctx,
+                                  problem.GetOutDataType(),
+                                  static_cast<uint32_t>(ck_args.N),
+                                  static_cast<uint32_t>(ck_args.K1),
+                                  static_cast<uint32_t>(ck_args.Do),
+                                  static_cast<uint32_t>(ck_args.Ho),
+                                  static_cast<uint32_t>(ck_args.Wo)};
+            }
+            else
+            {
+                return SolverType{ctx,
+                                  problem.GetOutDataType(),
+                                  static_cast<uint32_t>(ck_args.N),
+                                  static_cast<uint32_t>(ck_args.K1),
+                                  static_cast<uint32_t>(ck_args.Ho),
+                                  static_cast<uint32_t>(ck_args.Wo)};
```

Comment:
⚓ casting to uint32_t

Reply:
yes, this is a limit with the current transpose solver and kernel. We can add a TODO or open an issue for this.

Reply:
If this is out of scope of this PR, then let's open an issue. Otherwise it is better to fix right here, but I am not insisting.

[Note] `HasAtLeastOne64BitTensor()` has just been renamed to `AllTensorsDimsFitIntoInt()` in #2731, but the constraint is a bit relaxed here (`AllTensorsDimsFitIntoUint32()`?)

Reply:
[resolved]